links/src/commonMain/kotlin/com/splendo/kaluga/links/state/LinksStateRepo.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.links.state
+
+import co.touchlab.stately.concurrency.AtomicReference
+import com.splendo.kaluga.base.runBlocking
+import com.splendo.kaluga.links.Links
+import com.splendo.kaluga.links.manager.BaseLinksManager
+import com.splendo.kaluga.state.ColdStateRepo
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+
+class LinksStateRepo(
+    private val linksManagerBuilder: BaseLinksManager.Builder
+) : ColdStateRepo<LinksState>() {
+
+    interface Builder {
+        fun create(): LinksStateRepo
+    }
+
+    private val _linksManager: AtomicReference<BaseLinksManager?> = AtomicReference(null)
+    internal var linksManager: BaseLinksManager?
+        get() = _linksManager.get()
+        set(value) = _linksManager.set(value)
+
+    override suspend fun initialValue(): LinksState {
+        linksManager = linksManagerBuilder.create(::onLinksStateChange)
+
+        return LinksState.Pending
+    }
+
+    override suspend fun deinitialize(state: LinksState) {
+        linksManager = null
+    }
+
+    internal fun onLinksStateChange(link: Links) {
+        runBlocking {
+            takeAndChangeState { state ->
```

Comment:
All states go to all states, and states that go to themselves and repeat or do nothing. There is no special beha

In other words, it seems these are events, not states, that can be published on a Flowable.

It should be enough to publish these `Links` (`Link`?) to a Flowable (with a non-conflated `BroadcastChannel`). However in the next few days we'll introduce `SharedFlow`, it might be worth it to wait for that.